[Feat]-added hover effect and 'Read More' to blog cards

MadhavDhatrak opened the pull request:
<!-- In order to keep off topic discussion to a minimum, it helps if the "work to be done" is already agreed on. -->
<!-- Ideally, a GitHub Issue with the label `Status: Consensus` will have been concluded, and linked to in this PR. -->
<!--
Thanks for submitting a pull request! Please provide enough information so that others can review your pull request.
-->

**What kind of change does this PR introduce?**
Feature Addition- Add hover and Read More to blog cards

**Issue Number:**
<!-- Pick one of the below options.  Please remove those which don't apply. -->
-  Closes #1330 



**Screenshots/videos:**


https://github.com/user-attachments/assets/29b9b84f-10bd-425e-afd4-98eeafaf9be4



**If relevant, did you update the documentation?**
No

**Summary**
Added new UI enhancements to improve blog post interaction and readability:

- Added hover effects to the blog cards to make them visually interactive.
- Introducing a dedicated "Read More" button to guide users towards detailed blog posts.

**Does this PR introduce a breaking change?**

No

DhairyaMajmudar reviewed (changes requested):
@MadhavDhatrak if you check we're getting such background effect while hovering the headlines

[Screencast from 2025-01-31 23-47-18.webm](https://github.com/user-attachments/assets/c6e78ea8-aa13-4e22-9363-ec69f286700e)

Pls. fix them its not required

DhairyaMajmudar commented on styles/globals.css:
```diff
@@ -299,4 +299,31 @@ border-radius: 4px; */
+  animation: shine 2s infinite;
+}
+
+/* Add smooth transitions for dark mode */
+.dark .group:hover {
+  background: rgba(255, 255, 255, 0.05);
+}
+
+/* Enhance card interactions */
+.group {
+  transition: all 0.3s ease;
+}
+
+.group:hover {
```
Pls. remove comments too

DhairyaMajmudar commented:
Making blogs page a grid of 4xn makes it looks more airy, Pls. make it grid of 4 

![image](https://github.com/user-attachments/assets/1f72ecb7-18eb-4ca1-b1c8-54872145ab6e)

MadhavDhatrak commented:
Hey @DhairyaMajmudar 

As you requested I have made the following changes:

- Added a 4xN grid.
- Removed the hover background from headings.
- Removed comments from the codebase.


A video demo is attached below:

https://github.com/user-attachments/assets/cf27e958-5b12-4820-ab79-a064ed39e349

Please review the changes and let me know if any further modifications are needed.

benjagm commented:
This is looking mostly great but it is missing the last changes made on main. Can you please update from main and fix the merge conflicts to check if we can merge it?

DhairyaMajmudar reviewed (approved):
Awsm work, LGTM!

benjagm reviewed (changes requested):
Great job with this PR. When going to merge I found some conflicts I tried to fix to go and merge, but I found that the result for some cases didn't show the read more button. In mobile this is more frequent.

<img width="1260" alt="Screenshot 2025-03-16 at 10 32 48" src="https://github.com/user-attachments/assets/05b4bd86-f54c-483c-ad90-bcae8c491ada" />

MadhavDhatrak commented:
@benjagm  Some code for the "Read More" button may have been removed or overwritten during merge conflict resolution causing it to not appear in some cases No worries, I'll update it again!

MadhavDhatrak commented:
@benjagm  @DhairyaMajmudar 
I have resolved the merge conflicts, and the code is now working as expected. Please review it 
![image](https://github.com/user-attachments/assets/722cd0a7-5fff-4118-bf93-3770ec9b4562)

benjagm commented:
> I have resolved the merge conflicts, and the code is now working as expected. Please review it

This PR needs to include the changes added with #1477 to have all the blog images with the same size.

Utkarsh-123github commented:
Hi @MadhavDhatrak , as per Benjamin's comment: https://github.com/json-schema-org/website/issues/1330#issuecomment-2924813441 , I think we can close this issue. 
Thank you for your efforts and contribution🚀
